test/csuite/tiered_abort/main.c

```diff
@@ -0,0 +1,941 @@
+#define MIN_TH 5
+#define MIN_TIME 10
+#define NUM_INT_THREADS 3
+#define RECORDS_FILE "records-%" PRIu32
+/* Include worker threads and extra sessions */
+#define SESSION_MAX (MAX_TH + 4)
+#define WT_STORAGE_LIB "ext/storage_sources/local_store/.libs/libwiredtiger_local_store.so"
+
+static const char *table_pfx = "table";
+static const char *const uri_collection = "collection";
+static const char *const uri_local = "local";
+static const char *const uri_oplog = "oplog";
+static const char *const uri_shadow = "shadow";
+
+static const char *const sentinel_file = "sentinel_ready";
+
+static bool compat, inmem, use_ts;
+static volatile uint64_t global_ts = 1;
+static uint32_t flush_calls = 1;
+
+/*
+ * The configuration sets the eviction update and dirty targets at 20% so that on average, each
+ * thread can have a couple of dirty pages before eviction threads kick in. See below where these
+ * symbols are used for cache sizing - we'll have about 10 pages allocated per thread. On the other
+ * side, the eviction update and dirty triggers are 90%, so application threads aren't involved in
+ * eviction until we're close to running out of cache.
+ */
+#define ENV_CONFIG_ADD_COMPAT ",compatibility=(release=\"2.9\")"
+#define ENV_CONFIG_ADD_EVICT_DIRTY ",eviction_dirty_target=20,eviction_dirty_trigger=90"
+
+#define ENV_CONFIG_DEF                                        \
+    "cache_size=%" PRIu32                                     \
+    "M,create,"                                               \
+    "debug_mode=(table_logging=true,checkpoint_retention=5)," \
+    "eviction_updates_target=20,eviction_updates_trigger=90," \
+    "log=(archive=true,file_max=10M,enabled),session_max=%d," \
+    "statistics=(fast),statistics_log=(wait=1,json=true),"    \
+    "tiered_storage=(bucket=%s,bucket_prefix=pfx,name=local_store)"
+#define ENV_CONFIG_TXNSYNC \
+    ENV_CONFIG_DEF         \
+    ",transaction_sync=(enabled,method=none)"
+#define ENV_CONFIG_REC "log=(archive=false,recover=on)"
+
+/*
+ * A minimum width of 10, along with zero filling, means that all the keys sort according to their
+ * integer value, making each thread's key space distinct.
+ */
+#define KEY_FORMAT ("%010" PRIu64)
+
+typedef struct {
+    uint64_t absent_key; /* Last absent key */
+    uint64_t exist_key;  /* First existing key after miss */
+    uint64_t first_key;  /* First key in range */
+    uint64_t first_miss; /* First missing key */
+    uint64_t last_key;   /* Last key in range */
+} REPORT;
+
+typedef struct {
+    WT_CONNECTION *conn;
+    uint64_t start;
+    uint32_t info;
+} THREAD_DATA;
+
+/* Lock for transactional ops that set or query a timestamp. */
+static pthread_rwlock_t flush_lock;
+static pthread_rwlock_t ts_lock;
+
+static void handler(int) WT_GCC_FUNC_DECL_ATTRIBUTE((noreturn));
+static void usage(void) WT_GCC_FUNC_DECL_ATTRIBUTE((noreturn));
+static void
+usage(void)
+{
+    fprintf(stderr, "usage: %s [-h dir] [-T threads] [-t time] [-Cmvz]\n", progname);
```

Comment:
This prompted me to remove `Cm` support. I'm not sure where text would be useful for v, z.

Reply:
Ah.  `C` and `m` confused me a bit.  Hence this comment.  So this seems a bit less important now.  I had been thinking of adding this to the output of `usage()` Something like:

    usage: ...
        -v: verbose output
        -z: use timestamps

Reply:
To be fair, if that is desired, it should be a separate ticket because many csuite tests would have this exact same issue in slightly varying ways. Someone should consider what presentation is useful and consistently do that to all tests.

Reply:
That's fine.  To be honest, the problem I ran into was reading the code and trying to figure out what the `C` and `m` options did.  I initially looked for the `usage()` function, but it didn't say anything useful.  The current options are more obvious (for me anyway).  So I'm happy to drop this.  

I don't know if more extended usage output is needed in general for the csuite tests.